http/vertx/src/test/java/io/cloudevents/http/vertx/VertxMessageFactoryTest.java

```diff
@@ -33,9 +34,20 @@
 
 import static io.cloudevents.core.test.Data.*;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
 
 public class VertxMessageFactoryTest {
 
+    @Test
+    public void structuredMessageWithoutBody() {
+        MultiMap headers = MultiMap
+            .caseInsensitiveMultiMap()
+            .add("content-type", CSVFormat.INSTANCE.serializedContentType() + "; charset=utf8");
+
+        assertThatCode(() -> VertxMessageFactory.createReader(headers, null))
+            .hasMessageContaining("null body");
```

Comment:
Maybe check the type rather than the message?

Reply:
The type is CloudEventRWException, which wraps an illegal state exception... I can do it, not sure how much value it adds